### DISCUSSION TITLE
overwrite try parse method from primitive

### DISCUSSION THREAD

**Discussion by BlackBearFTW** [SteveDunn/Vogen#634] [Q&A] [answered]
Hi,

I would like to overwrite the existing tryparse methods that vogen hoists from the primitive so it supports some custom logic.

What am I trying to do: I am trying to support model binding. I am using sqid to display numeral ids as hashes, as a primitive I am using an integer since that is the persisted type in the database. However to the external world this should be shown as hash (like youtube). However when I try to bind my route to the value object, this of course doesn't work because I am trying to bind a string to a value object that uses an integer as primitive.

**Comment by SteveDunn** [comment]
Hi @BlackBearFTW - you can turn off hoisting and provide your own methods. I can assist if you could provide a minimal repo showing what needs to be done.

**Comment by SteveDunn** [comment, marked as answer]
You can now overwrite `Parse` and `TryParse`.  Vogen will see you've added your own and won't generate them.
